main.js

```diff
@@ -382,70 +390,108 @@ function createObjects(onConnect) {
+                                    compactProcs[procs[id].config.common.compactGroup].instances.splice(compactProcs[procs[id].config.common.compactGroup].instances.indexOf(id), 1);
+                                }
+                                if (procs[id].restartTimer) {
+                                    clearTimeout(procs[id].restartTimer);
+                                    delete procs[id].restartTimer;
+                                }
                                 delete procs[id];
                             }
                         });
                     } else {
-                        const __ipArr = getIPs();
-                        if (procs[id].config && (__ipArr.indexOf(procs[id].config.common.host) !== -1 || procs[id].config.common.host === hostname)) {
+                        const _ipArr = getIPs();
+                        if (procs[id].config && checkAndAddInstance(procs[id].config, _ipArr)) {
                             if (procs[id].config.common.enabled && (!procs[id].config.common.webExtension || !procs[id].config.native.webInstance)) {
                                 startInstance(id);
                             }
                         } else {
+                            // moved: also remove from instance list of compactGroup
+                            if (!compactGroupController && procs[id].config.common.compactGroup && compactProcs[procs[id].config.common.compactGroup] && compactProcs[procs[id].config.common.compactGroup].instances && compactProcs[procs[id].config.common.compactGroup].instances.includes(id)) {
+                                compactProcs[procs[id].config.common.compactGroup].instances.splice(compactProcs[procs[id].config.common.compactGroup].instances.indexOf(id), 1);
+                            }
+                            if (procs[id].restartTimer) {
+                                clearTimeout(procs[id].restartTimer);
+                                delete procs[id].restartTimer;
+                            }
                             delete procs[id];
                         }
                     }
-
                 } else if (obj && obj.common) {
                     const _ipArr = getIPs();
                     // new adapter
-                    if (_ipArr.indexOf(obj.common.host) !== -1 || obj.common.host === hostname) {
-                        procs[id] = {config: obj};
-                        if (procs[id].config.common.enabled && (!procs[id].config.common.webExtension || !procs[id].config.native.webInstance)) {
-                            startInstance(id);
-                        }
+                    if (checkAndAddInstance(obj, _ipArr) && procs[id].config.common.enabled && (!procs[id].config.common.webExtension || !procs[id].config.native.webInstance)) {
```

Comment:
This check also deserves a comment. I have no clue why all these conditions are necessary.

Reply:
Ok, they were that way before ... webExtensions and webInstance adapters do not have a process ... also not when disabled

Reply:
It was there before doesn't mean a comment is not necessary 😉